src/ngraph/runtime/gpu/gpu_memory_manager.hpp

```diff
@@ -75,10 +76,15 @@ namespace ngraph
                 std::vector<uint8_t> m_buffered_mem;
                 pass::MemoryManager m_workspace_manager;
                 static constexpr const uint16_t alignment = 8;
-                void* m_argspace;
-                void* m_workspace;
-                size_t m_allocation_size;
 
+                struct allocation
+                {
+                    void* ptr;
+                    size_t size;
+                };
+
+                std::list<allocation> m_argspace_mem;
```

Comment:
If my understanding is correct, we have a allocation for each function, is it better to use `std::unordered_map<string, allocation>`, where string is the unique function name. It will be more clear for `gpu_memory_manager` to handle.

Reply:
This would require us to correlate the reservation with the allocation. In order to do that, we would need to pass the unique function name as an argument to the GPUAllocator reservations.

Reply:
This is doable, but is it worth it? Would require a not insignificant API change to all the emitters in order to pass in the function name. Would also have to handle the case when there are multiple functions for a single external function.

Reply:
Just for the `m_argspace_mem` and `m_workspace_meme`. Let's not handle it for now. Currently it should be fine.

Reply:
> Just for the m_argspace_mem and m_workspace_meme

Yes, this is actually what I was referring to. Because GPUAllocator creates reservations which are eventually "looked up" in the lambda closure, the lookup key for a map would need to be available at the time of reservation so that it could be captured and later used in the primitive. It's doable, but it would require us to touch quite a bit more than just the argspace and workspace buffers.